### ISSUE TITLE
[Question]: Query Fails at exactly 23 Entity IDs in a list. What Setting To Increase Capacity?

### ISSUE THREAD

**Issue by johnshearing** [HKUDS/LightRAG#1656]
### Do you need to ask a question?

- [x] I have searched the existing question and discussions and this question is not already answered.
- [x] I believe this is a legitimate question, not just a bug or feature request.

### Your Question

Greetings all,
I am using a mix mode or local mode LightRAG query to look for orphans in the knowledge graph.

The following query works with any number of entities in the list up to 23:
The list of entities is generated by the http://localhost:9621/graph/label/list end point which is provided in the API tab of lightrag-server.

/mix [Only provide the name of the entity in the response. Nothing else is required.] Please examine the Entity ID for all Entities in the following python list. Then please return only the Entities with a Rank of 0. [ "2023-04-02T06:06:17Z", "2023-04-22T23:01:27Z", "Alabama", "Albury", "Biological Compartments", ... ]

Here is the response:

> Albury

This is correct, but if I add only one more entity to list of entities to be examined then I get the following failed response:

> I'm sorry, but I don't have the information regarding the Entity IDs or their ranks for the entities listed. Therefore, I cannot return the entities with a rank of 0.

So at the threshold of 24 items (where the query fails) I doubled the values of: Top K Results, Max Tokens for Text Unit, Max Tokens for Global Context, and Max Tokens for Local Context.
Unfortunately, I still get the same failed response.

I also tried increasing the following in the .env file but with no success.
```
entity_summary_to_max_tokens
max_token_size
chunk_token_size
```

If I put the same query into my python query script, then it will process many more list items than just 23.
I have not yet been able to determine what the limit is when using a python query script.

Can someone please tell me where the limit is specified for LightRAG Server, or how I can ask the question in a way which bypasses the limit?

Much thanks to all

**Comment by danielaskdd** [comment]
You cannot use RAG as a tool to query a long list of data.

 RAG is suitable for answering focused questions. RAG is not designed for comprehensive data enumeration or extracting extensive lists. This is determined by how RAG works. RAG only filters a limited number of content snippets from the knowledge base and sends them to the LLM to answer questions, making it unsuitable for generating lengthy data compilations in a single query.

**Comment by johnshearing** [comment]
Thanks @danielaskdd! 
Understood.

Now that I am starting to get familiar with the LightRAG server API I can use a python script to find all the orphans.
I will incorporate this function into the [merge application](https://github.com/johnshearing/deep_avatar/tree/main/LightRAG) I have been working on with the help of various a.i.
Soon we will have something to clean up dirty data and collect training data so as to fine-tune an a.i. to do the same.

I really appreciate your help.
